cdap-common/src/main/java/co/cask/cdap/common/metadata/AbstractMetadataClient.java

```diff
@@ -166,23 +172,23 @@ public MetadataSearchResponse searchMetadata(Id.Namespace namespace, String quer
    *              metadata from both {@link MetadataScope#SYSTEM} and {@link MetadataScope#USER}
    * @return The metadata for the entity.
    */
-  public Set<MetadataRecord> getMetadata(Id id, @Nullable MetadataScope scope)
+  public Set<MetadataRecord> getMetadata(EntityId id, @Nullable MetadataScope scope)
     throws NotFoundException, BadRequestException, UnauthenticatedException, IOException, UnauthorizedException {
 
-    if (id instanceof Id.Application) {
-      return getMetadata((Id.Application) id, scope);
-    } else if (id instanceof Id.Artifact) {
-      return getMetadata((Id.Artifact) id, scope);
-    } else if (id instanceof Id.DatasetInstance) {
-      return getMetadata((Id.DatasetInstance) id, scope);
-    } else if (id instanceof Id.Stream) {
-      return getMetadata((Id.Stream) id, scope);
-    } else if (id instanceof Id.Stream.View) {
-      return getMetadata((Id.Stream.View) id, scope);
-    } else if (id instanceof Id.Program) {
-      return getMetadata((Id.Program) id, scope);
-    } else if (id instanceof Id.Run) {
-      return getMetadata((Id.Run) id);
+    if (id instanceof ApplicationId) {
+      return getMetadata((Id.Application.fromEntityId((ApplicationId) id)), scope);
+    } else if (id instanceof ArtifactId) {
+      return getMetadata((Id.Artifact.fromEntityId((ArtifactId) id)), scope);
+    } else if (id instanceof DatasetId) {
+      return getMetadata((Id.DatasetInstance.fromEntityId((DatasetId) id)), scope);
+    } else if (id instanceof StreamId) {
+      return getMetadata((Id.Stream.fromEntityId((StreamId) id)), scope);
+    } else if (id instanceof StreamViewId) {
+      return getMetadata((Id.Stream.View.fromEntityId((StreamViewId) id)), scope);
+    } else if (id instanceof ProgramId) {
+      return getMetadata((Id.Program.fromEntityId((ProgramId) id)), scope);
+    } else if (id instanceof ProgramRunId) {
+      return getMetadata(Id.Run.fromEntityId((ProgramRunId) id));
```

Comment:
@bdmogal, why some of the methods in this class have an option to handle `RunId` while some doesn't ?